tests/test_io.py

```
@@ -427,23 +426,39 @@ def test_lzw_files(self):
     @pytest.mark.parametrize("extension", [".txt", ".bz2", ".gz", ".xz", ".lzma"])
-    def test_warnings(self, extension):
+    def test_warnings(self, extension, monkeypatch):
         filename = f"test_warning{extension}"
         content = "Test warning"
 
         with ScratchDir("."):
+            monkeypatch.setenv("PYTHONWARNDEFAULTENCODING", "1")
+
             # Default `encoding` warning
             with (
                 pytest.warns(EncodingWarning, match="use UTF-8 by default"),
```

Ruff (0.8.2)

Undefined name `EncodingWarning`. Consider specifying `requires-python = ">= 3.10"` or `tool.ruff.target-version = "py310"` in your `pyproject.toml` file.

(F821)